src/GraphQL/Execution/ExecutionHelper.cs

```diff
@@ -9,9 +10,34 @@ namespace GraphQL.Execution
     /// </summary>
     public static class ExecutionHelper
     {
+        private static readonly IDictionary<string, ArgumentValue> _emptyDirectiveArguments = new ReadOnlyDictionary<string, ArgumentValue>(new Dictionary<string, ArgumentValue>());
```

Comment:
Question 1: use empty inner dictionary or null ?

Reply:
50/50.  I'd say empty.  If someone wants to read arguments, it will save a null check.  If they don't attempt to read arguments, having the reference won't negatively affect anything.

Reply:
Ok